core/src/lib.rs

```diff
@@ -33,7 +35,8 @@ pub type Entid = i64;
 
 /// The attribute of each Mentat assertion has a :db/valueType constraining the value to a
 /// particular set.  Mentat recognizes the following :db/valueType values.
-#[derive(Clone,Copy,Debug,Eq,Hash,Ord,PartialOrd,PartialEq)]
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialOrd, PartialEq)]
+#[repr(u32)]
```

Comment:
It's in the "do this" section for `EnumSet`. I think it means "please represent this enum, which must be C-like — that is, with no associated values — as 32 bits, so I can be sure of my conversion to u32 later".

http://rustbyexample.com/custom_types/enum/c_like.html

A C-like enum gets a discriminant of 0, 1, 2, etc. for the cases.

`CLike::bit` then left-shifts 1 that many times.

This is equivalent to having a discriminant of 1, 2, 4, 8, 16, etc, which pack neatly into a bitfield. This is why we have a limitation of 31 cases in the enum: we only have 32 bit flags to flip.

Without forcing `repr(u32)` Rust might choose to use a single byte to represent this enum, potentially giving us junk or a crash when we do the unsafe `CLike::from_u32` inside `enum_set`.

Does that make sense?